.github/workflows/cifuzz.yml

```diff
@@ -0,0 +1,26 @@
+name: CIFuzz
+on: [pull_request]
+jobs:
+  Fuzzing:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Build Fuzzers
+      id: build
+      uses: google/oss-fuzz/infra/cifuzz/actions/build_fuzzers@master
+      with:
+        oss-fuzz-project-name: 'libsndfile'
+        dry-run: false
+        language: c
+    - name: Run Fuzzers
```

Comment:
I'm not sure we want to run the fuzzer in the pipeline, and having a 10 minute timeout seems excessive if we do. Also, the 10 minute timeout doesn't appear to have worked, looking at the CI jobs.

Reply:
@AdamKorcz , any ideas why timeout doesn't work?

Reply:
CC @jonathanmetzman

Reply:
So first, the timeout is only affects the time spent fuzzing which was 15 minutes instead of 10 minutes.
I tried investigating. It was hard to do so because of the amount these fuzzers write to stdout/stderr. I think it makes sense to stop this so that you can view stacktraces when the fuzzer finds crashes.

I was able to extract these logs that came from CIFuzz (rather than the fuzz targets):
```
2021-11-25T11:39:08.0814196Z 2021-11-25 11:39:08,080 - root - DEBUG - Is github: True.
2021-11-25T11:39:08.0820786Z 2021-11-25 11:39:08,081 - root - DEBUG - base_commit: None
2021-11-25T11:39:08.0871902Z 2021-11-25 11:39:08,081 - root - DEBUG - pr_ref: refs/pull/796/merge
2021-11-25T11:39:08.0873018Z 2021-11-25 11:39:08,081 - root - DEBUG - No PROJECT_SRC_PATH.
2021-11-25T11:39:08.0874201Z 2021-11-25 11:39:08,081 - root - INFO - ci_system: <continuous_integration.InternalGithub object at 0x7fac5c863df0>.
2021-11-25T11:39:08.0875760Z 2021-11-25 11:39:08,082 - root - INFO - ClusterFuzzDeployment: <clusterfuzz_deployment.OSSFuzz object at 0x7fac5c863f40>.
2021-11-25T11:39:08.0877068Z 2021-11-25 11:39:08,082 - root - INFO - InternalGithub: preparing for fuzzer build.
2021-11-25T11:39:08.0881276Z 2021-11-25 11:39:08,082 - root - INFO - Running: docker build -t gcr.io/oss-fuzz/libsndfile --file /opt/oss-fuzz/projects/libsndfile/Dockerfile /opt/oss-fuzz/projects/libsndfile.
2021-11-25T11:39:43.1736601Z 2021-11-25 11:39:43,172 - root - DEBUG - Stderr of command "git clone https://github.com/libsndfile/libsndfile.git libsndfile" is: Cloning into 'libsndfile'...
2021-11-25T11:39:43.3907587Z 2021-11-25 11:39:43,389 - root - DEBUG - Stderr of command "git fetch origin refs/pull/796/merge" is: From https://github.com/libsndfile/libsndfile
2021-11-25T11:39:43.4541954Z 2021-11-25 11:39:43,453 - root - DEBUG - Stderr of command "git checkout -f FETCH_HEAD" is: Note: switching to 'FETCH_HEAD'.
2021-11-25T11:39:43.4554705Z 2021-11-25 11:39:43,453 - root - INFO - repo_dir: /github/workspace/storage/libsndfile.
2021-11-25T11:39:44.2483757Z 2021-11-25 11:39:43,972 - root - INFO - Docker container: 70f867eeaf0c.
2021-11-25T11:39:44.2485167Z 2021-11-25 11:39:43,972 - root - INFO - Building with address sanitizer.
2021-11-25T11:39:44.2487783Z 2021-11-25 11:39:43,973 - root - INFO - Running: docker run --rm --privileged -e FUZZING_ENGINE=libfuzzer -e ARCHITECTURE=x86_64 -e CIFUZZ=True -e SANITIZER=address -e FUZZING_LANGUAGE=c -e OUT=/github/workspace/build-out --volumes-from 70f867eeaf0c gcr.io/oss-fuzz/libsndfile /bin/bash -c 'cd / && rm -rf /src/libsndfile/* && cp -r /github/workspace/storage/libsndfile /src && cd - && compile'.
2021-11-25T11:40:01.1634463Z checking for sysroot... no
2021-11-25T11:40:36.8147270Z 2021-11-25 11:40:36,813 - root - INFO - Removing unaffected fuzz targets.
2021-11-25T11:40:37.0237909Z 2021-11-25 11:40:37,022 - root - DEBUG - Diffing against base_ref: master.
2021-11-25T11:40:37.0240233Z 2021-11-25 11:40:37,022 - root - INFO - Diffing against master.
2021-11-25T11:40:37.0277781Z 2021-11-25 11:40:37,026 - root - INFO - Files changed in PR: ['.github/workflows/cifuzz.yml']
2021-11-25T11:40:37.3662862Z 2021-11-25 11:40:37,365 - root - INFO - Fuzz target sndfile_alt_fuzzer is affected by: ['ossfuzz/sndfile_alt_fuzzer.cc', 'ossfuzz/sndfile_fuzz_header.h', 'src/ALAC/ALACBitUtilities.c', 'src/ALAC/ag_dec.c', 'src/ALAC/alac_decoder.c', 'src/ALAC/dp_dec.c', 'src/ALAC/matrix_dec.c', 'src/ALAC/shift.h', 'src/G72x/g721.c', 'src/G72x/g723_24.c', 'src/G72x/g723_40.c', 'src/G72x/g72x.c', 'src/G72x/g72x_priv.h', 'src/GSM610/add.c', 'src/GSM610/decode.c', 'src/GSM610/gsm610_priv.h', 'src/GSM610/gsm_create.c', 'src/GSM610/gsm_decode.c', 'src/GSM610/gsm_destroy.c', 'src/GSM610/long_term.c', 'src/GSM610/rpe.c', 'src/GSM610/short_term.c', 'src/aiff.c', 'src/alac.c', 'src/alaw.c', 'src/au.c', 'src/audio_detect.c', 'src/avr.c', 'src/broadcast.c', 'src/caf.c', 'src/cart.c', 'src/chanmap.c', 'src/chunk.c', 'src/common.c', 'src/common.h', 'src/double64.c', 'src/dwvw.c', 'src/file_io.c', 'src/flac.c', 'src/float32.c', 'src/g72x.c', 'src/gsm610.c', 'src/htk.c', 'src/id3.c', 'src/ima_adpcm.c', 'src/ircam.c', 'src/mat4.c', 'src/mat5.c', 'src/mpc2k.c', 'src/mpeg.c', 'src/ms_adpcm.c', 'src/nist.c', 'src/nms_adpcm.c', 'src/ogg.c', 'src/paf.c', 'src/pcm.c', 'src/pvf.c', 'src/rf64.c', 'src/rx2.c', 'src/sds.c', 'src/sfendian.h', 'src/sndfile.c', 'src/strings.c', 'src/svx.c', 'src/txw.c', 'src/ulaw.c', 'src/voc.c', 'src/w64.c', 'src/wav.c', 'src/wavlike.c', 'src/wve.c', 'src/xi.c']
2021-11-25T11:40:37.3666759Z 2021-11-25 11:40:37,365 - root - INFO - Fuzz target sndfile_alt_fuzzer is not affected.
2021-11-25T11:40:37.4686195Z 2021-11-25 11:40:37,467 - root - INFO - Fuzz target sndfile_fuzzer is affected by: ['ossfuzz/sndfile_fuzz_header.h', 'ossfuzz/sndfile_fuzzer.cc', 'src/ALAC/ALACBitUtilities.c', 'src/ALAC/ag_dec.c', 'src/ALAC/alac_decoder.c', 'src/ALAC/dp_dec.c', 'src/ALAC/matrix_dec.c', 'src/ALAC/shift.h', 'src/G72x/g721.c', 'src/G72x/g723_24.c', 'src/G72x/g723_40.c', 'src/G72x/g72x.c', 'src/G72x/g72x_priv.h', 'src/GSM610/add.c', 'src/GSM610/decode.c', 'src/GSM610/gsm610_priv.h', 'src/GSM610/gsm_create.c', 'src/GSM610/gsm_decode.c', 'src/GSM610/gsm_destroy.c', 'src/GSM610/gsm_option.c', 'src/GSM610/long_term.c', 'src/GSM610/rpe.c', 'src/GSM610/short_term.c', 'src/aiff.c', 'src/alac.c', 'src/alaw.c', 'src/au.c', 'src/audio_detect.c', 'src/avr.c', 'src/broadcast.c', 'src/caf.c', 'src/cart.c', 'src/chanmap.c', 'src/chunk.c', 'src/common.c', 'src/common.h', 'src/double64.c', 'src/dwvw.c', 'src/file_io.c', 'src/flac.c', 'src/float32.c', 'src/g72x.c', 'src/gsm610.c', 'src/htk.c', 'src/id3.c', 'src/ima_adpcm.c', 'src/ircam.c', 'src/mat4.c', 'src/mat5.c', 'src/mpc2k.c', 'src/mpeg.c', 'src/ms_adpcm.c', 'src/nist.c', 'src/nms_adpcm.c', 'src/ogg.c', 'src/paf.c', 'src/pcm.c', 'src/pvf.c', 'src/rf64.c', 'src/rx2.c', 'src/sds.c', 'src/sfendian.h', 'src/sndfile.c', 'src/strings.c', 'src/svx.c', 'src/txw.c', 'src/ulaw.c', 'src/voc.c', 'src/w64.c', 'src/wav.c', 'src/wavlike.c', 'src/wve.c', 'src/xi.c']
2021-11-25T11:40:37.4700381Z 2021-11-25 11:40:37,467 - root - INFO - Fuzz target sndfile_fuzzer is not affected.
2021-11-25T11:40:37.4701342Z 2021-11-25 11:40:37,467 - root - INFO - No affected fuzz targets detected, keeping all as fallback.
2021-11-25T11:40:41.9117934Z 2021-11-25 11:40:41,911 - root - INFO - Build check passed.
2021-11-25T11:40:42.6180305Z 2021-11-25 11:40:42,617 - root - DEBUG - Is github: True.
2021-11-25T11:40:42.6194899Z 2021-11-25 11:40:42,619 - root - DEBUG - base_commit: None
2021-11-25T11:40:42.6196989Z 2021-11-25 11:40:42,619 - root - DEBUG - pr_ref: refs/pull/796/merge
2021-11-25T11:40:42.6197715Z 2021-11-25 11:40:42,619 - root - DEBUG - No PROJECT_SRC_PATH.
2021-11-25T11:40:42.6206221Z 2021-11-25 11:40:42,619 - root - INFO - Deleting builder docker images to save disk space.
2021-11-25T11:40:43.3962436Z 2021-11-25 11:40:43,394 - root - DEBUG - Stderr of command "docker rmi -f gcr.io/oss-fuzz/libsndfile gcr.io/oss-fuzz-base/base-builder gcr.io/oss-fuzz-base/base-builder-go gcr.io/oss-fuzz-base/base-builder-jvm gcr.io/oss-fuzz-base/base-builder-python gcr.io/oss-fuzz-base/base-builder-rust gcr.io/oss-fuzz-base/base-builder-swift" is: Error: No such image: gcr.io/oss-fuzz-base/base-builder-go
2021-11-25T11:40:43.4431637Z 2021-11-25 11:40:43,442 - root - INFO - ClusterFuzzDeployment: <clusterfuzz_deployment.OSSFuzz object at 0x7f7d18369670>.
2021-11-25T11:40:43.4433219Z 2021-11-25 11:40:43,442 - root - INFO - run fuzzers MODE is: code-change. Runner: <run_fuzzers.CiFuzzTargetRunner object at 0x7f7d18369880>.
2021-11-25T11:40:43.4438447Z 2021-11-25 11:40:43,443 - root - INFO - Using address sanitizer.
2021-11-25T11:40:43.4490823Z 2021-11-25 11:40:43,448 - root - INFO - Fuzz targets: ['/github/workspace/build-out/sndfile_alt_fuzzer', '/github/workspace/build-out/sndfile_fuzzer']
2021-11-25T11:40:43.4500110Z 2021-11-25 11:40:43,449 - root - INFO - Running fuzzer: sndfile_alt_fuzzer.
2021-11-25T11:40:43.4505748Z 2021-11-25 11:40:43,450 - root - INFO - Downloading corpus from OSS-Fuzz: https://storage.googleapis.com/libsndfile-backup.clusterfuzz-external.appspot.com/corpus/libFuzzer/libsndfile_sndfile_alt_fuzzer/public.zip
2021-11-25T11:40:46.9400156Z 2021-11-25 11:40:46,939 - root - INFO - Starting fuzzing
2021-11-25T11:45:58.6783822Z 2021-11-25 11:45:57,781 - root - INFO - Fuzzer sndfile_alt_fuzzer finished with no crashes discovered.
2021-11-25T11:45:58.6784996Z 2021-11-25 11:45:57,782 - root - INFO - Deleting corpus and seed corpus of sndfile_alt_fuzzer to save disk.
2021-11-25T11:45:58.6785910Z 2021-11-25 11:45:58,226 - root - INFO - Deleting fuzz target: sndfile_alt_fuzzer.
2021-11-25T11:45:58.6786910Z 2021-11-25 11:45:58,228 - root - INFO - Done deleting.
2021-11-25T11:45:58.6787763Z 2021-11-25 11:45:58,228 - root - INFO - Fuzzer sndfile_alt_fuzzer finished running without crashes.
2021-11-25T11:45:58.6788603Z 2021-11-25 11:45:58,228 - root - INFO - Running fuzzer: sndfile_fuzzer.
2021-11-25T11:45:58.6790558Z 2021-11-25 11:45:58,229 - root - INFO - Downloading corpus from OSS-Fuzz: https://storage.googleapis.com/libsndfile-backup.clusterfuzz-external.appspot.com/corpus/libFuzzer/libsndfile_sndfile_fuzzer/public.zip
2021-11-25T11:52:55.9272689Z 2021-11-25 11:45:59,500 - root - INFO - Starting fuzzing
2021-11-25T11:52:55.9273656Z 2021-11-25 11:51:09,792 - root - INFO - Fuzzer sndfile_fuzzer finished with no crashes discovered.
2021-11-25T11:52:55.9274688Z 2021-11-25 11:51:09,793 - root - INFO - Deleting corpus and seed corpus of sndfile_fuzzer to save disk.
2021-11-25T11:52:55.9275547Z 2021-11-25 11:51:10,018 - root - INFO - Deleting fuzz target: sndfile_fuzzer.
2021-11-25T11:52:55.9276280Z 2021-11-25 11:51:10,020 - root - INFO - Done deleting.
2021-11-25T11:52:55.9277079Z 2021-11-25 11:51:10,020 - root - INFO - Fuzzer sndfile_fuzzer finished running without crashes.
2021-11-25T11:52:55.9277980Z 2021-11-25 11:51:10,021 - root - INFO - Not uploading crashes because on OSS-Fuzz.
```
Fuzzing begins at `11:40:46` and ends at `11:51:10,021`, about 20 seconds after it was supposed to finish.
I don't know why it takes so long for github actions to finish after it seems like CIFuzz finishes.
Actually, I suspect the excessive logspam from the fuzz target has something to do with it. 
When I downloaded the logs, it comes out to about 500 MB.

So I think as the next step we can try disabling the logspam and if that doesn't eliminate the issue, I'll look into this further.
By the way, if you don't want to modify the target to fix this issue, you can put [`close_fd_mask`](https://llvm.org/docs/LibFuzzer.html#options) in .options files for the fuzz targets.

Reply:
I apologize for the delay in responding. I agree about spam in logs, it's worth removing it. Then we'll see how it will look.

Reply:
This should be fixed with https://github.com/google/oss-fuzz/pull/7004